lib/light-service/organizer/iterate.rb

```diff
@@ -11,12 +12,39 @@ def self.run(organizer, collection_key, steps)
           item_key = collection_key.to_s.singularize.to_sym
           collection.each do |item|
             ctx[item_key] = item
+
             ctx = scoped_reduce(organizer, ctx, steps)
+
+            next unless ctx.key?(:_rollback)
+
+            # Handle Rollback
+            rollback_items(
```

Comment:
I'm wondering (still not had the time to translate this wonder into a spec): if we're attaching `Iterate`'s rollback capability (`Iterate.rollback_items`) inside the reduce cycle what would happen if `fail_with_rollback!` will be called on the last iteration or after the `iterate` at all?

My hypothesis is that it will never been called, thus this logic is able to handle rollback only happened within the `collection.count - 1`th iteration cycle. Not on the last one nor if the rollback will be called from a later action outsite the `iterate`.

Am I missing something?